src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -380,9 +380,31 @@ private static void AssertFloatingPointBehavior<T>(T netfxExpectedValue, T netco
         public static void LongInputString(int length)
         {
             // Verify boundary conditions in Deserialize() that inspect the size to determine allocation strategy.
-            string repeated = new string('x', length - 2);
+            DeserializeLongJsonString(length);
+        }
+
+        // NOTE: VeryLongInputString test is constrained to run on Windows and MacOSX because it causes
+        //       problems on Linux due to the way deferred memory allocation works. On Linux, the allocation can
+        //       succeed even if there is not enough memory but then the test may get killed by the OOM killer at the
+        //       time the memory is accessed which triggers the full memory allocation.
+        [ConditionalTheory(nameof(IsX64))]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [InlineData(int.MaxValue - 3)]
```

Comment:
> Unfortuntately, I'm unable to run these tests locally due to OutOfMemoryException.

You can't new up strings of this length. I think using strings around the length of `int.MaxValue / 3` (i.e. `MaxExpansionFactorWhileTranscoding`) is reasonable for the tests.

You can't create strings of length larger than roughly `int.MaxValue/2 - 32`, i.e. 1_073_741_791
https://github.com/dotnet/coreclr/blob/ab2f9caaf35c96d029b96aa171ee65d04253cf7c/src/vm/gchelpers.cpp#L988-L989

Reply:
@ahsonkhan My bad, I'll fix it with upcoming commit. Should I also add test for the maximum possible string length (`int.MaxValue/2 - 32`)?

Reply:
> Should I also add test for the maximum possible string length

Sure, that's a good idea.

Please do try to run the tests locally to make sure they pass (either by removing the outerloop attribute on the test for a one-off run locally, or by running all outerloop tests - they take some time):
Navigate to the S.T.Json test project directly and run `dotnet msbuild /t:BuildAndTest /p:Outerloop=true`

Reply:
These tests pass on my machine. The only test that fail is `TestDateTimeOffset` but I think it happens because of my time zone settings (e.g. if I set time zone to `(UTC) Coordinated Universal Time` all tests pass).

Reply:
Great. Thanks @eugenesmlv.

Reply:
Can I add the test for the maximum possible string length in this PR or it would be better to create a separate pull request?

Reply:
Adding it here would be best. It's the same test/scenario.